### PR TITLE
configure spring security

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
Good job!

File: src/main/java/com/movie/store/configuration/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        List<String> errors = ex.getBindingResult()
+                .getFieldErrors()
+                .stream()
+                .map(x -> x.getDefaultMessage())
+                .collect(Collectors.toList());
+
+        body.put("errors", errors);
+
+        return new ResponseEntity<>(body, headers, status);
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/movie/store/configuration/CustomGlobalExceptionHandler.java]:
```suggestion
        body.put("status", status.value());
        List<String> errors = ex.getBindingResult()
                .getFieldErrors()
                .stream()
                .map(x -> x.getDefaultMessage())
                .collect(Collectors.toList());
        body.put("errors", errors);
        return new ResponseEntity<>(body, headers, status);
```

File: src/main/java/com/movie/store/configuration/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        auth
+                .inMemoryAuthentication()
+                .passwordEncoder(getEncoder())
+                .withUser("user")
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/movie/store/configuration/SecurityConfig.java]:
```suggestion
                .withUser("user")
```
does this user was registered via register() method and passed email validation?

File: src/main/java/com/movie/store/validation/EmailValidator.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements
+        ConstraintValidator<EmailConstraint, String> {
+    private static final String EMAIL_REGEX = "^(.+)@(.+)$";
+
+    @Override
+    public boolean isValid(String email,
+                           ConstraintValidatorContext cxt) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/movie/store/validation/EmailValidator.java]:
```suggestion
    public boolean isValid(String email, ConstraintValidatorContext cxt) {
```

File: src/main/java/com/movie/store/validation/PasswordConstraint.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Documented
+@Constraint(validatedBy = PasswordValidator.class)
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PasswordConstraint {
+    String message() default "Password has wrong format or dose not match to repeat password";
```
Comment (RomanDubovskyi) [inline src/main/java/com/movie/store/validation/PasswordConstraint.java:16]:
```suggestion
    String message() default "Password has wrong format or does not match to repeat password";
```

File: src/main/java/com/movie/store/controllers/UserController.java [hunk 5]
```diff
@@ -24,8 +24,8 @@ public UserController(UserService userService,
     }
 
     @GetMapping("/by-email")
-    public UserResponseDto getUserByEmail(@RequestParam String email) {
-        User user = userService.findByEmail(email).get();
+    public UserResponseDto getUserByEmail(Authentication authentication) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/movie/store/controllers/UserController.java:27]: you don't need to change this method, it should return user by given email is you send HTTP request

**Pull request merged**